client/my-sites/media-library/media-date-range.jsx

```diff
@@ -0,0 +1,344 @@
+		const now = new Date();
+
+		return (
+			<DatePicker
+				className="media-library__date-range-popover-date-picker"
+				showOutsideDays={ false }
+				toMonth={ now }
+				onSelectDay={ this.onSelectDate }
+				selectedDays={ {
+					from: this.momentDateToNative( this.state.startDate ),
+					to: this.momentDateToNative( this.state.endDate ),
+				} }
+				numberOfMonths={ window.matchMedia( '(min-width: 480px)' ).matches ? 2 : 1 }
+				calendarViewDate={ this.momentDateToNative( this.state.startDate ) }
+				disabledDays={ [
+					{
+						after: now, // you can't look at photos from the future!
+					},
+				] }
+			/>
+		);
+	}
+
+	renderPopoverHeader() {
+		return (
+			<div className="media-library__date-range-popover-header">
+				<Button
+					className="media-library__date-range-popover-cancel-btn"
+					onClick={ this.togglePopover }
+					compact
+				>
+					Cancel
+				</Button>
+				<Button
+					className="media-library__date-range-popover-apply-btn"
+					primary
+					onClick={ this.commitDates }
+					compact
+				>
+					<Gridicon icon="checkmark" />
+					Apply Dates
+				</Button>
+			</div>
+		);
+	}
+
+	renderTriggerButton() {
+		return (
+			<Button
+				className="media-library__date-range-btn"
+				ref={ this.startButtonRef }
+				onClick={ this.togglePopover }
+				compact
+			>
+				<Gridicon className="media-library__date-range-btn-icon" icon="calendar" />
+				<span className="media-library__date-range-btn-text">
+					{ this.dateToHumanReadable( this.state.startDate ) }-
+					{ this.dateToHumanReadable( this.state.endDate ) }
+				</span>
+				<Gridicon icon="chevron-down" />
+			</Button>
+		);
+	}
+
+	renderPopoverInputs() {
+		return (
+			<div className="media-library__date-range-popover-inputs">
+				<div className="media-library__date-range-popover-input media-library__date-range-popover-input--from">
+					<FormLabel htmlFor="startDate">From</FormLabel>
+					<FormTextInput
+						id="startDate"
+						name="startDate"
+						value={ this.state.inputFromDate }
+						onChange={ this.handleInputChange }
+						onBlur={ this.handleInputBlur }
+					/>
+				</div>
+				<div className="media-library__date-range-popover-input media-library__date-range-popover-input--to">
+					<FormLabel htmlFor="endDate">To</FormLabel>
+					<FormTextInput
+						id="endDate"
+						name="endDate"
+						value={ this.state.inputToDate }
+						onChange={ this.handleInputChange }
+						onBlur={ this.handleInputBlur }
+					/>
+				</div>
+			</div>
+		);
+	}
+
+	renderPopover() {
+		const popoverClassNames = classNames( {
+			'media-library__date-range-popover': true,
+			'is-dialog-visible': true, // forces to render when inside a modal. Is there a way to detect this from global state?
+		} );
+
+		return (
+			<Popover
+				className={ popoverClassNames }
+				isVisible={ this.state.popoverVisible }
+				context={ this.startButtonRef.current }
+				position="bottom"
+				onClose={ this.togglePopover }
```

Comment:
This ones more clear, but just to further the point - we should ask the question here: 'Should we be doing anything other than hiding the popover here (as in, the extra state that's being set in `toggleState`)

Reply:
Resolved in https://github.com/Automattic/wp-calypso/pull/29938